database/src/main/java/info/metadude/android/eventfahrplan/database/repositories/SessionsDatabaseRepository.kt

```diff
@@ -53,6 +84,39 @@ class SessionsDatabaseRepository(
         writableDatabase.delete(SessionByNotificationIdTable.NAME, SessionByNotificationIdTable.Columns._ID, "$notificationId")
     }
 
+    /**
+     * Updates or inserts sessions based on the given [contentValuesBySessionId].
+     */
+    fun upsertSessions(vararg contentValuesBySessionId: Pair</* sessionId */ String, ContentValues>) = with(sqLiteOpenHelper) {
+        writableDatabase.transaction {
+            contentValuesBySessionId.forEach {
+                upsertSession(it.first, it.second)
+            }
```

Comment:
```suggestion
            contentValuesBySessionId.forEach { (sessionId, contentValues) ->
                upsertSession(sessionId, contentValues)
            }
```

Reply:
OH! I am pretty sure I tried exactly this but the IDE/compiler did not let me do this. All in :-)

Reply:
You probably forgot the parenthesis. Doing so changes the signature of the lambda and then no `forEach` variant that takes such a function argument can be found.
If you think of the suggested snippet as a short form of the following it should become clearer of what is happening:

```kotlin
contentValuesBySessionId.forEach { value ->
    val (sessionId, contentValues) = value
    upsertSession(sessionId, contentValues)
}
```

The parenthesis are necessary for the destructuring.

Reply:
I remember I did that. Hm. It works now. I love it.

Reply:
:heavy_check_mark: Changed in af8c16aa9338397835d307ff029c21e9f2813532.